sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathClient.java

```diff
@@ -354,6 +362,140 @@ public PathInfo setPermissions(PathPermissions permissions, String group, String
         return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
     }
 
+    /**
+     * Sets the access control on a path and subpaths.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakePathClient.setAccessControlRecursive#List}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
+     *
+     * @param accessControlList The POSIX access control list for the file or directory.
+     * @return The result of the operation.
+     */
+    public AccessControlChangeResult setAccessControlRecursive(List<PathAccessControlEntry> accessControlList) {
+        return setAccessControlRecursiveWithResponse(new DirectorySetAccessControlRecursiveOptions(accessControlList), null,
+            Context.NONE).getValue();
+    }
+
+    /**
+     * Sets the access control on a path and subpaths.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakePathClient.setAccessControlRecursive#SetAccessControlRecursiveOptions-Duration-Context}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
+     *
+     * @param options {@link DirectorySetAccessControlRecursiveOptions}
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return A response containing the result of the operation.
+     */
+    public Response<AccessControlChangeResult> setAccessControlRecursiveWithResponse(
+        DirectorySetAccessControlRecursiveOptions options, Duration timeout, Context context) {
+        Mono<Response<AccessControlChangeResult>> response =
+            dataLakePathAsyncClient.setAccessControlRecursiveWithResponse(
+                PathAccessControlEntry.serializeList(options.getAccessControlList()), options.getProgressHandler(),
+                PathSetAccessControlRecursiveMode.SET, options.getBatchSize(), options.getMaxBatches(),
+                options.isContinuingOnFailure(), options.getContinuationToken(), context);
+
+        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
```

Comment:
Should we do the try catch error thing here so we can log it?

Reply:
Yea idk why we don't do it on all the datalake methods. I'll add it to all the other ones, too

Reply:
Hmm. Looks like we don't do that for any of the methods in BlobClientBase either, so I think it'd be better to do that as a separate PR